File: services/api/src/notification/notification.controller.ts
import { Controller, Get, Param, Put } from '@nestjs/common';
import { NotificationService } from './notification.service';
import { CurrentUser, AuthPrincipal } from '../common/decorators/current-user.decorator';

@Controller('notifications')
export class NotificationController {
  constructor(private readonly notificationService: NotificationService) {}

  @Get()
  async list(@CurrentUser() user: AuthPrincipal) {
    return this.notificationService.list(user.userId);
  }

  @Get('unread-count')
  async unreadCount(@CurrentUser() user: AuthPrincipal) {
    return this.notificationService.unreadCount(user.userId);
  }

  @Put(':id/read')
  async markRead(@CurrentUser() user: AuthPrincipal, @Param('id') id: string) {
    return this.notificationService.markRead(user.userId, id);
  }

  @Put('read-all')
  async markAllRead(@CurrentUser() user: AuthPrincipal) {
    return this.notificationService.markAllRead(user.userId);
  }
}

File: services/api-java-backup/src/main/java/com/launchly/notification/controllers/NotificationController.java
package com.launchly.notification.controllers;

import com.launchly.common.security.AuthContext;
import com.launchly.notification.entities.Notification;
import com.launchly.notification.services.NotificationService;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/api/notifications")
public class NotificationController {
    private final NotificationService notificationService;

    public NotificationController(NotificationService notificationService) {
        this.notificationService = notificationService;
    }

    @GetMapping
    public ResponseEntity<List<Notification>> list() {
        return ResponseEntity.ok(notificationService.listByUser(AuthContext.userId()));
    }

    @GetMapping("/unread-count")
    public ResponseEntity<Map<String, Long>> unreadCount() {
        return ResponseEntity.ok(Map.of("count", notificationService.unreadCount(AuthContext.userId())));
    }

    @PutMapping("/{id}/read")
    public ResponseEntity<Void> markRead(@PathVariable String id) {
        notificationService.markAsRead(id);
        return ResponseEntity.ok().build();
    }

    @PutMapping("/read-all")
    public ResponseEntity<Void> markAllRead() {
        notificationService.markAllAsRead(AuthContext.userId());
        return ResponseEntity.ok().build();
    }
}
